static/js/stream_data.js

```diff
@@ -799,13 +799,52 @@ exports.sort_for_stream_settings = function (stream_ids) {
         return sub.name.toLocaleLowerCase();
     }
 
-    function by_stream_name(id_a, id_b) {
+    function weekly_traffic(stream_id) {
+        const sub = exports.get_sub_by_id(stream_id);
+        if (sub && sub.is_old_stream) {
+            return sub.stream_weekly_traffic;
+        } else {
+            // don't intersperse new streams with zero-traffic existing streams
+            return -1;
+        }
+    }
+
+    function by_stream_name(id_a, id_b)  {
         const stream_a_name = name(id_a);
         const stream_b_name = name(id_b);
         return String.prototype.localeCompare.call(stream_a_name, stream_b_name);
     }
 
-    stream_ids.sort(by_stream_name);
+    function by_subscriber_count(id_a, id_b) {
+        const out = exports.get_sub_by_id(id_b).subscribers.size
+        - exports.get_sub_by_id(id_a).subscribers.size;
```

Comment:
I'm concerned that this is going to be extremely expensive if you have 1000 subscribers in a stream, because we call this comparison function many times during a sort.

@showell what are your thoughts on how to handle this this -- is the right model to just cache on the `sub` object the size of the subcriber lists?

Reply:
FWIW, unless I'm misunderstanding something about how `subs.js` works, I'm pretty sure this perf issue already exists in master. The [existing `stream_data.sort_for_stream_settings` function](https://github.com/zulip/zulip/blob/master/static/js/stream_data.js#L799-L818) uses the same pattern, and is called via `subs.filter_table` and `get_stream_id_buckets` every time a user changes the search filter.

Reply:
Hmm, OK.  

I think we should definitely fix this, but I suppose we can do it afterwards.  If you're interested in the project, I can talk through what's involved.

@showell ping on this.

Reply:
`subscribers.size` should already be O(1), so this particular sort should be the normal O(NlogN)